hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/iskyc/IsKycCall.java

```diff
@@ -0,0 +1,74 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.iskyc;
+
+import static com.hedera.hapi.node.base.ResponseCodeEnum.INVALID_ACCOUNT_ID;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.SUCCESS;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult.successResult;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.defaultfreezestatus.DefaultFreezeStatusTranslator.DEFAULT_FREEZE_STATUS;
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.accountNumberForEvmReference;
+import static java.util.Objects.requireNonNull;
+
+import com.esaulpaugh.headlong.abi.Address;
+import com.hedera.hapi.node.base.ResponseCodeEnum;
+import com.hedera.hapi.node.state.token.Token;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.HederaSystemContract.FullResult;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.AbstractNonRevertibleTokenViewCall;
+import com.hedera.node.app.service.contract.impl.hevm.HederaWorldUpdater;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+public class IsKycCall extends AbstractNonRevertibleTokenViewCall {
+    private final Address account;
+
+    public IsKycCall(
+            @NonNull final HederaWorldUpdater.Enhancement enhancement,
+            @Nullable final Token token,
+            @NonNull final Address account) {
+        super(enhancement, token);
+        this.account = requireNonNull(account);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected @NonNull HederaSystemContract.FullResult resultOfViewingToken(@NonNull final Token token) {
+        requireNonNull(token);
+        // TODO - gas calculation
+
+        final var accountNum = accountNumberForEvmReference(account, nativeOperations());
+        if (accountNum < 0) {
+            return fullResultsFor(INVALID_ACCOUNT_ID, 0L, false);
+        }
+        var tokenRel = nativeOperations()
+                .getTokenRelation(accountNum, token.tokenIdOrThrow().tokenNum());
+        var result = tokenRel == null ? false : tokenRel.kycGranted();
+        return fullResultsFor(SUCCESS, 0L, result);
+    }
+
+    @Override
+    protected @NonNull FullResult viewCallResultWith(@NonNull ResponseCodeEnum status, long gasRequirement) {
+        return fullResultsFor(status, gasRequirement, false);
+    }
+
+    private @NonNull FullResult fullResultsFor(@NonNull ResponseCodeEnum status, long gasRequirement, boolean isKyc) {
+        return successResult(
+                DEFAULT_FREEZE_STATUS.getOutputs().encodeElements(status.protoOrdinal(), isKyc), gasRequirement);
```

Comment:
🪳 `DEFAULT_FREEZE_STATUS` -> `IS_KYC`? 

(Maybe others should be checked too ... I'm not going backwards to find out, sorry)

Reply:
yikes - thanks for catching this.  I'll double check the other call functions.  Too much copy/pasting of code.